### PR TITLE
Optimize bottlenecks

### PR DESCRIPTION
**Pull request by michaeleisel**
Here are some bottlenecks that I measured to have a significant perf impact. They're a little rough right now, but if they look good overall I can clean them up

### CODE REVIEW

File: Sources/PathKit.swift [hunk 1]
```diff
@@ -227,7 +227,9 @@ extension Path {
   /// - Returns: the last path component
   ///
   public var lastComponent: String {
-    return NSString(string: path).lastPathComponent
+    var component = NSString(string: path).lastPathComponent
+    component.makeContiguousUTF8()
```
Comment (michaeleisel) [inline Sources/PathKit.swift]: it's important to call `makeContiguousUTF8` if it's going to be used much later, e.g. for string comparison. it allows swift to use fast paths in the string code if possible

**Comment by michaeleisel** [comment]
friendly ping. i'd love to keep moving on this while it's still fresh in my head.

By the way, one big question is, how to deal with the C file? SwiftPM doesn't support mixed-language projects, and other package managers have issues as well. This is the first non-Swift file, right? Here are some options:

- Translate the C to Swift where it makes extensive use of unsafe pointers. This may incur some performance overhead
- Create a separate package, e.g. XcodeProjCExt, whose version is synced with that of XcodeProj. This is what I do for https://github.com/michaeleisel/ZippyJSON, which puts its non-Swift code in https://github.com/michaeleisel/ZippyJSONCFamily . This is a pain, but also a good long-term investment if it will be good to have more code like this in the future. XcodeProj is used in large projects where it has a large workload, so I think it could be worth it. Just as a thought experiment, if the whole project were written in C++ (which I am not actually suggesting), it could easily be 10x faster. This is what I'm doing in-progress for XcodeProj at https://github.com/tuist/XcodeProj/pull/529
- ?

**Comment by michaeleisel** [comment]
s/XcodeProjCExt/PathKitCExt

**Comment by michaeleisel** [comment]
Is this project still maintained?

**Comment by RomanPodymov** [comment]
Hello @michaeleisel 
First of all, you should fix Travis CI errors.

**Pull request closed without merging**